Tools/GNUMake/comps/gnu.mak

```diff
@@ -114,12 +114,14 @@ endif
 
 ########################################################################
 
-ifeq ($(gcc_major_version),4)
-  CXXFLAGS += -std=c++11
-else ifeq ($(gcc_major_version),5)
-  CXXFLAGS += -std=c++14
+ifeq ($(shell expr $(gcc_major_version) \< 5),1)
+  ifeq ($(shell expr $(CXXSTD) \> 11),1)
+    $(error C++14 support requires GCC 5 or newer.)
+  endif
 endif
-CFLAGS     += -std=gnu99
+
+CXXFLAGS += -std=c++$(CXXSTD)
+CFLAGS   += -std=gnu99
```

Comment:
not sure how well gnu C extensions and iso c++ mix in corner cases...

Reply:
This seems to change the make system behavior.  Currently for gcc >= 6, we use whatever the default of gcc is.  But now, we use C++11 by default.

Reply:
Yep, we can also keep the default in `Tools/GNUMake/Make.defs` empty and keep that logic, as you prefer.

Reply:
If we want to keep the old behaviour, I would implement blocks like this in every compiler makefile:
```make
ifdef CXXSTD
  CXXSTD := $(strip $(CXXSTD))
  ifeq ($(shell expr $(gcc_major_version) \< 5),1)
    ifeq ($(shell expr $(CXXSTD) \> 11),1)
      $(error C++14 support requires GCC 5 or newer.)
    endif
  endif
  CXXFLAGS += -std=c++$(CXXSTD)
else
  ifeq ($(gcc_major_version),4)
    CXXFLAGS += -std=c++11
  else ifeq ($(gcc_major_version),5)
    CXXFLAGS += -std=c++14
  endif
endif
```

Reply:
The documentation entry for users should then read for the current behavior:
```rst
+------------|-------------------------------------+--------------------+
| CXXSTD     | C++ standard (``c++11``, ``c++14``) | compiler default,  |
|            |                                     | at least ``c++11`` |
+------------+-------------------------------------+--------------------+
```

Reply:
Yes, I think that's the way to go.  Should be `11` instead of `c++11`, right?

Reply:
Okay with me.
`11` vs. `c++11`: https://github.com/AMReX-Codes/amrex/pull/630#discussion_r354539768